ss58-registry.json

```diff
@@ -73,6 +73,15 @@
       "standardAccount": "*25519",
       "website": "https://bifrost.finance/"
     },
+    {
+      "prefix": 6,
```

Comment:
You can not reuse an index. Do you even need this?

Reply:
Hello @bkchr  We have two networks, one is Polkadot, and one is Kusama, and they have the same Token (BNC). Prefix are both 6, if not, we can not call our ledger app in Ledger wallet (we also have two networks Ledger app)

Reply:
If you have two different networks, you should use different indices. Everything else doesn't make any sense. This index is also used to have different addresses per network to protect users.

Reply:
> If you have two different networks, you should use different indices. Everything else doesn't make any sense. This index is also used to have different addresses per network to protect users.

Here is a question, Bifrost Kusama and Bifrost Polkadot are keeping the same prefix, but if we use different indices on extension, then users will see different address format on the same network.

Reply:
@bkchr As you can see, we are using the same prefix:6 on Bifrost Polkadot and Kusama:
Bifrost Polkadot Metadata: https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Fhk.p.bifrost-rpc.liebi.com%2Fws#/settings/metadata
Bifrost Kusama Metadata: https://polkadot.js.org/apps/?rpc=wss%3A%2F%2Fbifrost-parachain.api.onfinality.io%2Fpublic-ws#/settings/metadata

Reply:
Each chain should have its own prefix. These prefixes are not unique per currency.

Reply:
> Each chain should have its own prefix. These prefixes are not unique per currency.

But seems there is no restriction in Runtime prefix, should we change it or keep it still?

Reply:
In addition, for those projects were already deploying to both Kusama and Polkadot parachains, like Astar, Phala, KILT, HydraDX and including Bifrost, seems they all using the same prefix on both networks, so how should the same prefix be proceed in the future?

Reply:
> no restriction in Runtime prefix

What is the runtime prefix?

Reply:
And the ss58 prefix never hits the runtime. This is only for the "outside"

Reply:
> And the ss58 prefix never hits the runtime. This is only for the "outside"

Sorry, I am not a dev, but I meant this in the capture below, its parachain prefix right?
<img width="713" alt="image" src="https://user-images.githubusercontent.com/72777624/187115952-82f45e5a-ac22-4f8c-83c9-5e22b25d100b.png">

The thing is we are currently connecting with Leager team, but seems now ss58 can only display one network format in two of the same parachain prefix. For example, in the list of "Display Address Format For" I can only select Bifrost Polkadot but there is no Bifrost Kusama.

Thus, Leager can only support Bifrost Polkadot OR Bifrost Kusama, rather than both of them. Seems other networks have the same prefix will face the same issue. 
<img width="296" alt="image" src="https://user-images.githubusercontent.com/72777624/187116201-05877507-82df-43b1-b655-19a8de81ad78.png">

Reply:
> Seems other networks have the same prefix will face the same issue.

The point here is that no other network is doing this, because this isn't supported. This prefix is used to make the addresses unique per chain. The idea is that your UI should prevent sending tokens to some address X on chain B, while address X belongs to chain A.

Reply:
> 

okey, understood, but one more question, why there are still some parachain prefix on kusama same as their Polkadot network? We are one of them, both prefix 6, what we can do with this to make sure its "unique".

Reply:
@jacogr can you help here?

Reply:
The linked screenshot for display just means (in this case), use "6". Generally you won't need to touch that, that is a global setting applying to all for format display. If you have a network added by "metadata update" you can tie an account on a per-network basis.

Reply:
Addendum: The extension certainly doesn't have support for "Ledger Bifrost Kusama" at all. Support for that has been added 19 days ago in the common libs (here https://github.com/polkadot-js/common/tree/v10.1.4) but the extension doesn't magically update itself. 

It needs a build and publish to the stores and there has not been a release to the stores for a while (judging by current progress, the next store submission is probably > 1 month away, assuming the hanging issues for next update is addressed). Unlike stuff likes the apps UI, there is no "publish on merge" it is a completely manual submission process with store reviews, resubmissions, etc. So it doesn't get taken lightly and with at least a 2 day+ review cycle each submissions, it certainly is not done each time there are new libraries on npm, which is weekly.

Reply:
> The linked screenshot for display just means (in this case), use "6". Generally you won't need to touch that, that is a global setting applying to all for format display. If you have a network added by "metadata update" you can tie an account on a per-network basis.
> 
> The Ledger app specifically works on genesisHash and Ledger app id, it has nothing to do with the ss58 prefix which is only for human display.

We are trying to add Leager in both of our networks (kusama and polkadot), but Leager cannot successfully match both of them because the ss59-registry only registered our Polkadot nerwork. So here is why we stuck here:
- We got same parachain prefix on both network, which are “6”;
- We need both networks match Leager app;
- We’ve already added bifrost-kusama at _polkadot-js-common/packages/networks/src/defaults/genesis.ts_, then we have to register bifrost-kusama on _paritytech/ss58-registry/blob/main/ss58-registry.json_, but the prefix cannot be the same.
- Only if our both network can be registered [here](https://github.com/paritytech/ss58-registry/blob/main/ss58-registry.json), so leader can sync two networks’ genesisHash.

Reply:
I'll try and explain again.

1. The ss58 prefix does nothing to do with Ledger - picking the app based on the genesisHash of the chain (and then each app has an id)
2. The extension will not work with Ledger on Kusama until a new version hits the stores (while the library has support, the extension at this point does not use that version)

Reply:
As Jaco explained, this shouldn't be an issue for your Ledger integration. Merging this PR that duplicates a prefix isn't an option so here is another idea if you feel that there is no other way to move forward and you really want to add both chains to the SS58 registry.

The Bifrost Polkadot parachain is in a very good spot right now. You have all the filters enabled to prevent balances transfer and any other type of transaction. You also have Sudo in your parachain so making a runtime upgrade is fast and straight forward.

The required fixes are on your side and you need to consider this other option if you are completely blocked.

You need to change the SS58 prefix for you Polkadot parachain. It is a very easy chage in your runtime and in your JSON spec file.
In this file `runtime/bifrost-polkadot/src/lib.rs` you need to change the `pub const SS58Prefix: u8 = 6;` to a new and unused index.
https://github.com/bifrost-finance/bifrost/blob/c43c1964fc7ba13d9085862ed048ab4c875cd049/runtime/bifrost-polkadot/src/lib.rs#L186

And in your JSON spec file `node/service/res/bifrost-polkadot.json` change `"ss58Format": 6` to a new unused SS58 index.
https://github.com/bifrost-finance/bifrost/blob/c43c1964fc7ba13d9085862ed048ab4c875cd049/node/service/res/bifrost-polkadot.json

Then you just need to do a runtime upgrade and your new index can be added to the SS58 registry.